core/src/saros/concurrent/management/ResourceActivityFilter.java

```diff
@@ -0,0 +1,263 @@
+package saros.concurrent.management;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+import org.apache.log4j.Logger;
+import saros.activities.DeletionAcknowledgmentActivity;
+import saros.activities.FileActivity;
+import saros.activities.FileActivity.Type;
+import saros.activities.IActivity;
+import saros.activities.IResourceActivity;
+import saros.activities.SPath;
+import saros.filesystem.IProject;
+import saros.session.AbstractActivityConsumer;
+import saros.session.IActivityConsumer;
+import saros.session.IActivityConsumer.Priority;
+import saros.session.ISarosSession;
+import saros.session.ISessionListener;
+import saros.session.User;
+
+/** Class to handle file deletions and filter out resource activities for already deleted files. */
+class ResourceActivityFilter {
+  private static final Logger log = Logger.getLogger(ResourceActivityFilter.class);
+
+  private final ISarosSession sarosSession;
+
+  /** Method passed by the constructor caller that can be used to react to file deletions. */
+  private final Consumer<SPath> fileDeletionHandler;
```

Comment:
There is only one method implementation. Any reason not pass the ConcurrentDocumentServer and call the specific method instead ?

Reply:
This class is created for both the [`ConcurrentDocumentClient`](https://github.com/saros-project/saros/blob/570e6310f088fe1d13da50e2db90c0ddb4db7e44/core/src/saros/concurrent/management/ConcurrentDocumentClient.java#L40) as well as the [`ConcurrentDocumentServer`](https://github.com/saros-project/saros/blob/570e6310f088fe1d13da50e2db90c0ddb4db7e44/core/src/saros/concurrent/management/ConcurrentDocumentServer.java#L64). The two classes are using different methods, which is why I generalized this.